src/Microsoft.Diagnostics.DebugServices.Implementation/DataReader.cs

```diff
@@ -202,7 +205,7 @@ public override ulong GetExportSymbolAddress(string symbol)
                 return 0;
             }
 
-            public override IResourceNode ResourceRoot => base.ResourceRoot;
+            public override IResourceNode ResourceRoot => _resourceRoot ??= ModuleInfo.TryCreateResourceRoot(_reader, _module.ImageBase, _module.ImageSize, _module.IsFileLayout.GetValueOrDefault(false));
```

Comment:
It might be a good idea to wrap this with a try/catch.

The `PEImage` code that backs TryCreateResourceRoot isn't designed to throw exceptions, but it's not hardened.  If you throw random data at it (or this gets called on a Mach-O or Elf image), you might get unexpected exceptions.

We would certainly fix those as we see them, but it wouldn't hurt to do some defensive coding here.

Reply:
(And a Trace/log so we know an exception unexpectedly occurred.)

Reply:
I did look through the code that TryCreateResourceRoot calls (mostly constructors) and saw no exceptions being thrown exception ArgumentException.  My philosophy is exceptions like that or any other ones that might happen should perk up though to the handler the host has around the command loop.  

I usually don't just catch Exception either exception in a few cases like in a reverse-pinvoke function that native C++ called.

Reply:
I think this code is similar (without any try/catches) to clrmd's PEModuleInfo.Resources property.

Reply:
This change looks good then.